projects/Douglas Mentoring/day1.py

```diff
@@ -0,0 +1,92 @@
+import getpass
+import netmiko
+# json was here bc as I'm using textfsm to pull data from cisco devices
+# it comes in json i needed to show the output before deciding how to tread the date
+import json
+import csv
+
+# Where i'm pulling the devices list / Soon I'll make this automated which will pull direct from NARF
+file = open(r'/home/segattod/segattod-workspace/src/Segattod-package/myproject/Light_level_checks/list.txt',
+            'r')
+file = file.read().splitlines()
```

Comment:
This is generally a bad idea.  below you used the `with` context manager like this:

```
with open(file, "r"):
    stuff....
```

And that is the "right" way.   One of the main problems with this way here is that you never closed the file.  If someone else tried to open that file even if you aren't currently running your script, they could run into errors because it's locked by you.   The operating system should eventually realize your script is done and eventually unlock it, but it's a bad idea to cross your fingers and hope.

The context manager magically closes the file when the code leaves the manager (hits the unindent) so that is never an issue.


Which generally also means it should be inside a function or method.  But you already know that :P

Reply:
I actually didn't know it was a best practice just felt right to move it to a variable and work with that after(LOL)

You mentioned "the context manager magically closes .." do you mean the way i did it, is good, it will always close automatically after "moving" the content to the variable, right?

just confirming if it go it right lol